compiler/rustc_span/src/span_encoding.rs

```diff
@@ -193,7 +193,11 @@ impl Span {
             // the interned value contains all the data, so we don't need to
             // distinguish them.
             let index = self.lo_or_index;
-            with_span_interner(|interner| interner.spans[index as usize])
+            let mut data = with_span_interner(|interner| interner.spans[index as usize]);
+            if data.ctxt.as_u32() == u32::MAX {
```

Comment:
Same sentinel value used here, should we add a helper, something like `data.ctxt.is_sentinel()`?

Reply:
This should probably use `self.ctxt_or_parent_or_marker == CTXT_INTERNED_MARKER` instead, so the value in the interned span data is never read.